### PR TITLE
#1553 Update Lambda Dependencies

### PR DESCRIPTION
**Pull request by EvanParish**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

This change removes the lambda `requirements.txt` files and creates a readme with a table that says which layers are used by which lambdas. There was also a small modification to the lambda deploy action because a lambda layer was added for PyJWT[crypto] so that no longer needs to be manually imported by the lamda.

issue #1553 

## Type of change

Please check the relevant option(s).

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Documentation changes only

## How Has This Been Tested?

How has this been tested? (e.g. Unit tests, Tested locally, Tested as a GitHub action, Depoyed to dev, etc)  
Please provide relevant information and / or links to demonstrate the functionality or fix. (e.g. screenshots, link to deployment, regression test results, etc)

- [x] [Deployed to Dev](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/7427249980)

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an appropriate title: #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/release.yaml) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have ensured the latest master is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] The ticket is now moved into the DEV test column

### CODE REVIEW

**Review by edDocMe360 — DISMISSED** [review]

File: lambda_functions/readme.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Lambdas and the Layers they use
```
Comment (cris-oddball) [inline lambda_functions/readme.md:1]: do we need any info here on the need to pin urllib3 until we can move to python 3.10?

**Review by cris-oddball — CHANGES_REQUESTED** [review]

File: lambda_functions/readme.md (hunk 1 above)
Comment (EvanParish) [reply lambda_functions/readme.md:1]: Good point. I'll add a comment about that.

**Review by cris-oddball — APPROVED** [review]

File: lambda_functions/readme.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+| pinpoint_callback_lambda |  |
+| pinpoint_inbound_sms_lambda |  |
+| ses_callback_lambda |  |
+| two_way_sms_v2 | psycopg2-binary, requests |
+| va_profile_opt_in_out_lambda | psycopg2-binary, pyjwt |
+| va_profile_remove_old_opt_outs_lambda | psycopg2-binary |
+| vetext_incoming_forwarder_lambda | twilio |
+
+**Note:** _PyJWT Layer_  
+Whenever PyJWT[crypto] (pyjwt-layer) is updated, we also must update the pyjwt requirement in api `requirements_for_test.txt` file.
```
Comment (kalbfled) [inline lambda_functions/readme.md:19]: There is a similar comment in requirements-for-test.txt that should also be updated.

**Review by kalbfled — CHANGES_REQUESTED** [review]

File: lambda_functions/readme.md (hunk 2 above)
Comment (EvanParish) [reply lambda_functions/readme.md:20]: good point, I'll update that as well.

**Review by k-macmillan — APPROVED** [review]

**Review by cris-oddball — APPROVED** [review]

**Review by kalbfled — APPROVED** [review]

**Pull request merged**